gammapy/modeling/models/core.py

```diff
@@ -612,6 +669,61 @@ def selection_mask(
 
         return np.array(selection, dtype=bool)
 
+    def contribute(self, mask, psf=None):
+        """Check if a model contribute within a mask map.
+           Note that BackgroundModel are assumed to always contribute. 
+    
+        Parameters
+        ----------
+        mask : `~gammapy.maps.WcsNDMap` of boolean type
+            Map containing a boolean mask
+        psf : `~gammapy.irf.PSFKernel` or `~gammapy.irf.PSFMap`
+            PSF kernel
+        
+        Returns
+        -------
+        inside : `numpy.array` of boolean type
+            True where models contribute within the mask geom
+            and its center is located inside the region where mask==True
+        outside : `numpy.array` of boolean type
+            True  where models contribute within the mask geom
+            but its center is located ouside the region where mask==True
+        """
+
+        nmod = len(self)
+        inside = np.zeros(nmod, dtype=bool)
+        outside = np.zeros(nmod, dtype=bool)
+        for k, m in enumerate(self):
+            inside[k], outside[k] = m.contribute(mask, psf)
+        return inside, outside
```

Comment:
Here I would just return one of them and add the `inside=True/False` parameter. I can see how it can be advantageous to return both, however for API consistency with `.region_mask()` and `.boundary_mask()` I would prefer the `inside=` option.